packages/wallets/src/components/Base/WalletTextField/WalletTextField.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ChangeEvent, InputHTMLAttributes, ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import WalletText from '../WalletText/WalletText';
+import styles from './WalletTextField.modules.css';
+
+interface WalletTextFieldProps extends InputHTMLAttributes<HTMLInputElement> {
+    defaultValue?: string;
+    helperMessage?: string;
+    icon?: ReactElement;
+    label?: string;
+    leftIcon?: React.ReactNode;
+    showMessage?: boolean;
+}
+
+const WalletTextField: React.FC<WalletTextFieldProps> = ({
+    defaultValue = '',
+    helperMessage,
+    icon,
+    label,
+    showMessage = false,
+    ...rest
+}) => {
+    const [value, setValue] = useState(defaultValue);
```

Comment:
I'm always keen on minimising the amount of state to upkeep and manage - would it  be reasonable to have flow, where parent component provides current value (as it needs to keep it anyway) and thus WalletTextField does not need to duplicate the state?

Reply:
There are a few variations for the Input component. To allow them to work out of the box, I have to include some states there. i.e. Character Limit